### ISSUE TITLE
Button clicks on Android

### ISSUE THREAD

**Issue by Kiraffi** [KennyShields/LoveFrames#162]
I tried using LoveFrames with android device, and I noticed that the button clicks do work, but it requires to double click on button to get it to work. The first press I think is needed to set cursor on top of button and second click to activate it. I was wondering if it is somehow possible to set hover and check button click in one go or simply ignore whole hovering thing and simply check button clicks? 

Now the first click sets the hovering on and second one then checks if its hovered and does the action.

**Comment by KennyShields** [comment]
Yeah, the best solution would probably be to remove the hover checks.

**Comment by Ladyink1313** [comment]
Need help

**Comment by KennyShields** [comment]
What do you need help with exactly? Changing the buttons to work on android devcies?

**Comment by qaisjp** [comment]
personally I think this is actually a flaw in the android port of LoveFrames - mobile devices (with touchscreens) don't have cursors so I feel that the android port should actually just send the onclick event whenever the screen is pressed (whilst updating getCursorPosition).

I doubt anything could be done here if the user is dragging the cursor across the screen. I haven't tried Love2d on android yet though.

**Comment by karai17** [comment]
Any ideas for a definitive fix for this? I am releasing a game on Android that uses loveframes and this bug is really, really getting on my nerves.

**Comment by KennyShields** [comment]
I don't have anything at the moment, but I can probably come up with a solution when I'm less busy. When are you planning to release your game?

**Comment by karai17** [comment]
Yesterday! https://play.google.com/store/apps/details?id=com.binarycocoa.humbleboco

**Comment by karai17** [comment]
I have a fix:

```lua
function newobject:is_binding(x, y)
	if not self.visible then return end

	local ex, ey = self:GetPos()
	local ew, eh = self:GetSize()

	if ex <= x and ex + ew >= x and ey <= y and ey + eh >= y then
		return true
	end

	return false
end
```

Basically an AABB check to see if the click position is within the element's limits. I replaced the hover check with this.
